src/main/kotlin/io/github/intellij/dlanguage/actions/CreateDlangClassAction.kt

```diff
@@ -29,26 +29,29 @@ import javax.swing.Icon
  * Created by pirocks on 9/21/16.
  * mostly copy-pasted from "CreateDlangFileAction.java"
  */
-open class CreateDlangClassAction : CreateFileFromTemplateAction(NEW_D_FILE, "", DLanguage.Icons.CLASS), DumbAware {
+class CreateDlangClassAction : CreateFileFromTemplateAction(NEW_D_FILE, "", DLanguage.Icons.CLASS), DumbAware {
 
     override fun buildDialog(project: Project, directory: PsiDirectory, builder: CreateFileFromTemplateDialog.Builder) {
+        builder.setTitle(NEW_D_FILE)
+
         for (template in Template.entries) {
             with(template) {
                 builder.addKind(StringUtil.capitalize(readableName), icon, id)
             }
         }
 
-        builder.setTitle(NEW_D_FILE)
-            .setValidator(ClassNameValidator())
+        builder.setValidator(ClassNameValidator)
     }
 
     override fun createFileFromTemplate(name: String, template: FileTemplate, inRequestedDirectory: PsiDirectory): PsiFile {
         val sourceRoot = findSourceRootOrDefault(inRequestedDirectory, inRequestedDirectory.virtualFile)
 
+        val inputName = if (name.endsWith(".d")) name.dropLast(2) else name
+
         val segments = inRequestedDirectory.virtualFile.path.removePrefix(sourceRoot.path).split("/")
             .filter { it.isNotBlank() }
             .toMutableList()
-        val nameSegments = name.split(".")
+        val nameSegments = inputName.split(".")
```

Comment:
The core of why this PR, I was annoyed of having a module named "d.d" when I create test files for the plugin and I put the ".d" extension in the name.